### DISCUSSION TITLE
Using static libraries instead of dynamic ones

### DISCUSSION THREAD

**Discussion by Bolderaysky** [xmake-io/xmake#4871] [Q&A]
When `add_requires` used to download package from xrepo, it does install both dynamic and static libraries in the same directory in specific cases, and as far as I see, xmake passes paths to such directories to the linker, which means linker will choose dynamic libraries. Is there any way to change that, so libraries will be linked statically? I can't pass `-static` to the linker as there are certain libraries I can't link statically.

Minimal xmake configuration for better understanding:
```lua
add_requires("liburing", {system = false})

target("test")
    set_kind("binary")
    add_files("src/*.cpp")
    add_packages("liburing") -- liburing is linked dynamically here
```

**Comment by waruqi** [comment]
you can open a pr to improve this package. like this https://github.com/xmake-io/xmake-repo/blob/52e84d5aa56cf9e3ce66f167fe82df586461c0a9/packages/p/pcre/xmake.lua#L41

it should be static library by default.
